conan/tools/gnu/autotoolstoolchain.py

```diff
@@ -120,11 +133,97 @@ def defines(self):
         ret = [self.ndebug, self.gcc_cxx11_abi] + conf_flags + self.extra_defines
         return self._filter_list_empty_fields(ret)
 
+    def _get_cc(self):
+        return self._exe_env_var_to_unix_path(
+            "CC",
+            "cl" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+            self._compile_wrapper,
+        )
+
+    def _get_cxx(self):
+        return self._exe_env_var_to_unix_path(
+            "CXX",
+            "cl" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+            self._compile_wrapper,
+        )
+
+    def _get_ld(self):
+        return self._exe_env_var_to_unix_path(
+            "LD",
+            "link" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+        )
+
+    def _get_ar(self):
+        return self._exe_env_var_to_unix_path(
+            "AR",
+            "lib" if is_msvc(self._conanfile) else None,
+            ["-nologo"] if is_msvc(self._conanfile) else [],
+            self._arlib_wrapper,
+        )
+
+    def _get_nm(self):
+        return self._exe_env_var_to_unix_path(
+            "NM",
+            "dumpbin" if is_msvc(self._conanfile) else None,
+            ["-nologo", "-symbols"] if is_msvc(self._conanfile) else [],
+        )
+
+    def _get_objdump(self):
+        return self._exe_env_var_to_unix_path(
+            "OBJDUMP",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _get_ranlib(self):
+        return self._exe_env_var_to_unix_path(
+            "RANLIB",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _get_strip(self):
+        return self._exe_env_var_to_unix_path(
+            "STRIP",
+            ":" if is_msvc(self._conanfile) else None,
+        )
+
+    def _exe_env_var_to_unix_path(self, env_var, default=None, extra_options=[], wrapper=None):
```

Comment:
Using a list as default keyword value in Python is dangerous.
The list is not modified in the body, but might be in later iterations.
It's safer to set the default value to `None` and then set the parameter to an empty list in the body.

Many Python linters warn about this.

Reply:
https://github.com/conan-io/conan/pull/12194/commits/f03d98dbdf0e399f0555416f5de7eece3d52fcd4